integration/integration_plugins_test.go

```diff
@@ -0,0 +1,66 @@
+// +build integration
+
+package main
+
+import (
+	delugeclient "github.com/gdm85/go-libdeluge"
+	"testing"
+	"time"
+)
+
+func waitForPluginEnabled(t *testing.T, name string) {
+	tick := time.NewTicker(time.Millisecond * 500)
+	defer tick.Stop()
+
+	for attempt := 0; attempt < 10; attempt ++ {
+		// Sleep before testing enabled plugins.
+		<-tick.C
+
+		t.Logf("Attempt %d waiting for plugin %s to become enabled", attempt + 1, name)
+
+		plugins, err := c.GetEnabledPlugins()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, p := range plugins {
+			if p == name {
+				return
+			}
+		}
+	}
+
+	t.Fatalf("Timeout waiting for plugin %s to become enabled", name)
+}
+
+func testWithPlugin(t *testing.T, name string) func() {
+	err := deluge.EnablePlugin(name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	waitForPluginEnabled(t, name)
+
+	return func() {
+		err := deluge.DisablePlugin(name)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// cleanup DebugServerResponses after plugin disable
+		c.DebugServerResponses = nil
+	}
+}
+
+func TestLabelPluginGetLabels(t *testing.T) {
+	defer testWithPlugin(t, "Label")()
```

Comment:
The only part in `defer` should be the disable, not the enabling.

I am not sure how this can even work since the enabling is happening when the test finishes.

Reply:
Defer is actually calling the function returned by `testWithPlugin` (hence the double sets of brackets).

The question suggests that isn't readable enough. An attempt at DRY with the plugin name. I'll simplify it.

Reply:
Is that better?

Reply:
Thanks, impressive work!